app/src/test/kotlin/org/cru/godtools/ui/dashboard/lessons/LessonsViewModelTest.kt

```diff
@@ -62,9 +95,50 @@ class LessonsViewModelTest {
         val second = randomTool("second", Tool.Type.LESSON, defaultOrder = 2, isHidden = false)
 
         viewModel.lessons.test {
-            lessonsFlow.value = listOf(second, first)
+            lessonsByLanguageFlow.value = listOf(second, first)
             runCurrent()
             assertEquals(listOf("first", "second"), expectMostRecentItem())
         }
     }
+
+    @Test
+    fun `Property lessons - Filter by language`() = testScope.runTest {
+        val first = randomTool("first", Tool.Type.LESSON, defaultOrder = 1, isHidden = false)
+        val second = randomTool("second", Tool.Type.LESSON, defaultOrder = 2, isHidden = false)
+
+        viewModel.lessons.test {
+            viewModel.updateSelectedLanguage(Language(Locale.ENGLISH))
+            lessonsFlow.value = listOf(first, second)
+            lessonsByLanguageFlow.value = listOf(first)
+            appLanguagesFlow.value = Locale.ENGLISH
+            languageFlow.value = listOf(Language(Locale.ENGLISH))
+            runCurrent()
+            assertEquals(listOf("first"), expectMostRecentItem())
+        }
+    }
+
+    @Test
+    fun `Property filteredLanguages - Filter languages`() = testScope.runTest {
+        val first = Language(Locale.ENGLISH)
+        val second = Language(Locale.FRENCH)
+
+        viewModel.filteredLanguages.test {
+            languageFlow.value = listOf(first, second)
```

Comment:
set the value of the `languageFlow` before you start testing `filteredLanguages`, then you shouldn't need to use `runCurrent()` and can using `awaitItem()` instead of `expectMostRecentItem()`